doc/layer2-forwarding-enhancements/SONiC Layer 2 Forwarding Enhancements HLD.md

```diff
@@ -0,0 +1,444 @@
+- Spanning Tree Protocol  to flush dynamic FDB entries either on a Port or on (Port,VLAN) when topology change occurs. API will be provided in Orchagent for L2 protocol component to flush FDB entries accordingly.
+  
+
+**2. Handle MAC move events.**
+- MAC move event generated by some hardware (e.g. DNX) is currently not handled in SAI and SONiC. Will add support in Orchagent for moving the FDB entry to new port upon receipt of such event. 
+  
+
+**3. Configuration for FDB aging time.**
+- CLI configuration  to be added to change the FDB aging time. By default, the aging time in hardware is 0. It can be changed to a desired value if CLI is available. Currently, it can be done only by setting in APP_DB. The configuration range will be  0-1,000,000 seconds. Setting aging time to 0 will disable aging.
+  
+
+**4. Configuration for static FDB entry.**
+- Currently, static FDB can only be added only by setting in APP_DB. CLI configuration will be added for this.
+- If a dynamic FDB already exists with the same (MAC, VLAN), it will be replaced with the static entry.
+- If the port is not member of VLAN, the static entry will be present in CONFIG_DB and APP_DB. It will also be saved in the orchagent saved FDB until the port is added to the VLAN.
+  
+
+**5. FDB clear options for per port and per VLAN ans per Port per VLAN clear.**
+
+ - Currently, "sonic-clear fdb" command has only the option "ALL" supported. Will add options for "PORT" and "VLAN" to clear FDB entries either on a port or on a VLAN or on PORT and VLAN. Only dynamic FDB will be cleared.
+
+ **6. VLAN Range CLI.**
+- In current SONiC release, support for VLAN commands through CLI is limited to creating and deleting a single VLAN and adding/removing a port to/from a single VLAN. Support for VLAN range will be added in CLI to enable administrator to create and delete a range of VLANs and add/remove a port to/from a range of VLANs.
+- When a large number of interfaces are made members of a huge set of VLANs, deletion of VLANs takes significant time due to the reason mentioned in [Scalability](#7-scalability) section.
+-  In SONiC, VLAN operations are performed either through config_db.json file or through CLI commands. In config_db.json file, administrator is required to update the desired VLAN and/or member port in each VLAN as per the JSON file format. This process is overburdening when it is required to manually add/delete VLANs in bulk and add/delete ports to/from VLANs in bulk. Administrator can perform the same operations through CLI commands which, though, is easy, becomes extremely taxing when multiple commands need to be executed to perform bulk operations, like creating/deleting 4094 VLANs and associating/removing ports from each of these VLANs through CLI commands that provide an option to enter only one VLAN.
+ - Administrator's task is greatly simplified with provision for VLAN range configuration. And time taken to perform the operations is drastically reduced.
```

Comment:
vlan range feature is great addition, how do you handle/propagate config validation errors to the users? will it be chatty or aggregated error/warning messages?  Prefer to have an option to the user what kind of error propagation? aggregated or not?

Reply:
A warning message is logged if the -w option is used. Please refer to section 3.1